nixos/modules/services/networking/nncp.nix

```diff
@@ -0,0 +1,151 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  statefulCfgFile = "/etc/nncp.secrets";
+  nixosCfgFile = "/run/nncp.hjson";
+  programCfg = config.programs.nncp;
+  daemonCfg = config.services.nncp.daemon;
+  settingsFormat = pkgs.formats.json { };
+  jsonCfgFile = settingsFormat.generate "nncp.json" programCfg.settings;
+in {
+  options = {
+    programs.nncp = {
+
+      enable = mkEnableOption "NNCP (Node to Node copy) utilities";
+
+      group = mkOption {
+        type = types.str;
+        default = "nncp";
+        description = ''
+          UNIX group that users must be in to use <literal>nncp</literal>.
+        '';
+      };
+
+      settings = mkOption {
+        type = settingsFormat.type;
+        description = ''
+          NNCP configuration, see
+          <link xlink:href="http://www.nncpgo.org/Configuration.html"/>.
+          At runtime these settings will overlay the contents of the file
+          <literal>${statefulCfgFile}</literal> and will be symlinked at
+          <literal>/etc/nncp.hjson</literal>. If <literal>${statefulCfgFile}</literal>
+          does not exist then it will be generated.
+          Node keypairs go in <literal>${statefulCfgFile}</literal>, do
+          not specify them in <literal>settings</literal> as they will be leaked
+          into <literal>/nix/store</literal>!
+        '';
+        default = { };
+      };
+
+    };
+
+    services.nncp = {
+
+      daemon = {
+        enable = mkEnableOption ''
+          NNCP TCP synronization daemon.
+          The daemon will take configuration from
+          <xref linkend="opt-programs.nncp.settings"/>
+        '';
+        listenStreams = mkOption {
+          type = with types; listOf str;
+          description = ''
+            TCP sockets to bind to.
+            See <xref linkend="opt-systemd.sockets._name_.listenStreams"/>.
+          '';
+          default = [ "5400" ];
+        };
+        user = mkOption {
+          type = types.str;
+          description = ''
+            UNIX user to run the daemon as.
+            What permissions the daemon requires depends on
+            whether <literal>autotoss</literal> is enabled and which
+            commands are enabled for remote execution.
+          '';
+          default = "nobody";
+        };
+        extraArgs = mkOption {
+          type = with types; listOf str;
+          description = "Extra command-line arguments to pass to daemon.";
+          default = [ ];
+          example = [ "-autotoss" ];
+        };
+      };
+
+    };
+  };
+
+  config = mkIf (programCfg.enable or daemonCfg.enable) {
+
+    programs.nncp.settings = {
+      spool = mkDefault "/var/spool/nncp";
+      log = mkDefault "/var/spool/nncp/log";
```

Comment:
This is a traditional [spool directory](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s14.html) and I would not move it. It would contradict the spirit of NNCP as a UUCP replacement as well as the documentation.